crates/optimism/payload/src/builder.rs

```diff
@@ -389,8 +389,8 @@ impl<Txs> OpBuilder<'_, Txs> {
         let executed: ExecutedBlock<N> = ExecutedBlock {
             recovered_block: Arc::new(block),
             execution_output: Arc::new(execution_outcome),
-            hashed_state: Arc::new(hashed_state),
-            trie_updates: Arc::new(trie_updates),
+            hashed_state: Arc::new(hashed_state.into_sorted()),
```

Comment:
This is a complication that I hadn't thought about... that we might be making block building slower by forcing the builder to do the sorting itself, rather than the validator. There's a couple of possibilities here:

* Use an Either for hashed_state/trie_updates fields on ExecutedBlock, so they can be sorted or not.
* Use a different ExecutedBlock struct for block building, which doesn't seem wrong but might be a big change
* Benchmark block building and see what the affect is

cc @mattsse